icicle/tests/test_hash_api.cpp

```diff
@@ -315,54 +322,216 @@ class HashSumBackend : public HashBackend
   }
 };
 
-void assert_valid_tree(
+/**
```

Comment:
why do you need doxygen documentation for tests? also the doc string is wrong (missing leaf-size).

Reply:
Because it's not the test itself but a function used by the tests so I added documentation to it
I'll add leaf size to the doc